accelerator/abstract_accelerator.py

```diff
@@ -287,3 +287,7 @@ def build_extension(self):
     @abc.abstractmethod
     def export_envs(self):
         ...
+
+    @abc.abstractmethod
+    def get_accelerator_backend(self, device_index=None):
```

Comment:
1. Compile backend should be global rather than per device. 
2. We are returning compilation backend, so `compile_backend` is a more apt name. 
```suggestion
    def get_compile_backend(self):
```